app/models/answer.rb

```diff
@@ -0,0 +1,41 @@
+class Answer < ActiveRecord::Base
+  has_many :comments, as: :commentable
+  has_many :votes, as: :votable
+  belongs_to :question, touch: true
+  belongs_to :responder, class_name: 'User'
+  delegate :username, to: :responder, allow_nil: true
+
+  def editable_by? user
+    responder == user
+  end
+
+  def my_comments
```

Comment:
You don't need this. answer.comments is at least as comprehensible as answer.my_comments